File: examples/governance_static_go/domain/renewed_training.go
package domain

import "time"

type RenewedTraining struct {
	AggregateID string    `json:"aggregate_id"`
	AggregateId string `json:"aggregate_id"`
	TrainingRecordId string `json:"training_record_id"`
	Certification string `json:"certification"`
	ExpiresAt time.Time `json:"expires_at"`
	StakeholderId string `json:"stakeholder_id"`
	PolicyId string `json:"policy_id"`
	CompletedAt time.Time `json:"completed_at"`
	Status string `json:"status"`
	OccurredAt time.Time `json:"occurred_at"`
}

func (e RenewedTraining) EventName() string { return "RenewedTraining" }

func (e RenewedTraining) GetOccurredAt() time.Time { return e.OccurredAt }

File: examples/governance/compliance_domain/lib/compliance_domain/training_record/events/renewed_training.rb
module ComplianceDomain
  class TrainingRecord
    module Events
      class RenewedTraining
        attr_reader :aggregate_id, :training_record_id, :certification, :expires_at, :stakeholder_id, :policy_id, :completed_at, :status, :occurred_at

        def initialize(aggregate_id: nil, training_record_id: nil, certification: nil, expires_at: nil, stakeholder_id: nil, policy_id: nil, completed_at: nil, status: nil)
          @aggregate_id = aggregate_id
          @training_record_id = training_record_id
          @certification = certification
          @expires_at = expires_at
          @stakeholder_id = stakeholder_id
          @policy_id = policy_id
          @completed_at = completed_at
          @status = status
          @occurred_at = Time.now
          freeze
        end
      end
    end
  end
end
